controllers/admin/self-managed/LogsController.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ */
+
+namespace PrestaShop\Module\AutoUpgrade\Controller;
+
+use InvalidArgumentException;
+use PrestaShop\Module\AutoUpgrade\AjaxResponseBuilder;
+use PrestaShop\Module\AutoUpgrade\Task\TaskType;
+use PrestaShop\Module\AutoUpgrade\Twig\PageSelectors;
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+class LogsController extends AbstractGlobalController
+{
+    public function getDownloadLogsButton(): JsonResponse
+    {
+        $type = $this->request->request->get('download-logs-type');
+        $this->validateLogsType($type);
+
+        return AjaxResponseBuilder::hydrationResponse(
+            PageSelectors::DOWNLOAD_LOGS_PARENT_ID,
+            $this->getTwig()->render(
+                '@ModuleAutoUpgrade/components/download_logs.html.twig',
+                $this->upgradeContainer->getLogsService()->getDownloadLogsdData($type)
+            )
+        );
+    }
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    private function validateLogsType(string $type): void
+    {
+        if (!in_array($type, TaskType::ALL_TASKS)) {
+            throw new InvalidArgumentException('Unknown log type ' . $type);
+        }
+    }
+}
```

Comment:
It could go into the log service, right?

Reply:
Yeah I thought of it too. But the methods of `LogsService` are already typed as `@param TaskType::TASK_TYPE_* $task`, meaning we've have checked the value before calling it. I think we have two options here:
* The current implementation,
* A new validator for this type of variable.

Reply:
ok I understand. I just find it a shame to have to find this kind of function in a controller because of a typing problem. Let's leave it like that

Reply:
Updated in favor of a check in TypeTask class